block-header-gossip.md

```diff
@@ -0,0 +1,198 @@
+
+
+## Block storage
+Each client will be storing blocks in two forms. As accumulators where the entire block history is recorded and as partial block headers where only a subset of the block header is stored. Partial block headers are only stored for the most recent N number of blocks.
+
+### Accumulator
+Portal network's accumulator will be based on the [double-batched merkle log accumulator](https://ethresear.ch/t/double-batched-merkle-log-accumulator/571) 
+
+#### Epoch Accumulator
+A fixed sized accumulator of a proposed length of 2048. Block info will be added in into the SSZ list until it has filled up all 2048 of its entries. After which the next block will be included in a new epoch accumulator.
+
+SSZ sede structure:
+`List[Container[blockhash:bytes32, total_difficulty:uint256], max_length=2048]`
+
+The root hash of an epoch will then be included as an entry in the master accumulator.
+
+Aside from the first epoch, each client will need to store the current epoch as well as the epoch before in case of reorgs and block sync.
+
+#### Master Accumulator
+
+An accumulator that increases in size with each epoch. The root hash of each epoch will be appended to the master accumulator
+
+SSZ sede structure:
+`List[epoch_root_hash:bytes32, max_length=<a_large_number>]`
+
+
+### Partial Block Header
+
+Clients will need more information to verify the validity of the block. In the event of network latency, some clients may have missed out on the last few blocks and will need to request for the partial block header from its neighbours. 
+
+And so, each client number will also be required to store a certain amount of the most recent blocks.
+
+Partial Block Header will include:
+- Block Number
+- Block Hash
+- Difficulty
+- Nonce
+- Mixhash
+- Previous Hash (may be required to determine longest, heaviest chain. to know if the current block is pointed to the previous block)
+
+## DHT
+The block header gossip will be an overlay of [DiscV5](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-theory.md).
+
+#### <u>Idea No.1</u>
+It is the responsibility of all clients in the portal network to have a local version of accumulator and block header. The block header gossip can reuse the ENR from the underlying DiscV5 protocol. The maintenence of node liveness and node records are being done solely by DiscV5.
+
+Information that are required by Block header gossip can be passed onto the overlay block header gossip layer in TalkReq/TalkResp encapsulated message.
+
+Pros:
+1. Able to reuse underlying DiscV5
+2. Pass the responsibility of node maintence to DiscV5 and reduce additional message exchange
+
+Cons:
+1. Could be potentially too tightly coupled
+
+
+#### <u>Idea No.2</u>
+A seperate DHT for ENR's are being maintained in the overlay network. Similar messages such as PING, PONG, FINDNODES and FOUNDNODES will be used to maintained node liveness in the DHT. 
```

Comment:
This is the option we want to use.  I am also fine with authoring this part (the wire protocol) of the spec.  It should generally mirror how it has been declared in the state/history/tx-gossip documents.

Reply:
have removed Idea No.1 and moved the option that we are using into the wire protocol section